opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py

```diff
@@ -212,3 +217,44 @@ def _get_opentelemetry_stability_opt_in(
             return _OpenTelemetrySemanticConventionStability._OTEL_SEMCONV_STABILITY_SIGNAL_MAPPING.get(
                 type, _OpenTelemetryStabilityMode.DEFAULT
             )
+
+
+def is_instrumentation_enabled() -> bool:
+    if context.get_value(_SUPPRESS_INSTRUMENTATION_KEY):
+        return False
+    return True
+
+
+def is_http_instrumentation_enabled() -> bool:
+    return (
+        is_instrumentation_enabled()
+        and not
+        context.get_value(_SUPPRESS_HTTP_INSTRUMENTATION_KEY)
+    )
+
+
+@contextmanager
+def _suppress_instrumentation(*keys: str) -> Iterable[None]:
+    """Suppress instrumentation within the context."""
+    ctx = context.get_current()
+    for key in keys:
+        ctx = context.set_value(key, True, ctx)
+    token = context.attach(ctx)
+    try:
+        yield
+    finally:
+        context.detach(token)
+
+
+@contextmanager
+def suppress_instrumentation() -> Iterable[None]:
+    """Suppress instrumentation within the context."""
+    with _suppress_instrumentation(_SUPPRESS_INSTRUMENTATION_KEY):
+        yield
```

Comment:
We might want to also set the `"suppress_instrumentation"` (with no uuid) key here for any users that were relying on it.

Reply:
We kept these suppress keys internal so far so it should be fine to not set that.